apps/openmw/mwmechanics/spelllist.cpp

```diff
@@ -0,0 +1,160 @@
+#include "spelllist.hpp"
+
+#include <components/esm/loadspel.hpp>
+#include <components/misc/rng.hpp>
+
+#include "spells.hpp"
+
+#include "../mwbase/environment.hpp"
+#include "../mwbase/world.hpp"
+
+#include "../mwworld/esmstore.hpp"
+
+namespace
+{
+    template<class T>
+    const std::vector<std::string> getSpellList(const std::string& id)
+    {
+        return MWBase::Environment::get().getWorld()->getStore().get<T>().find(id)->mSpells.mList;
+    }
+
+    template<class T>
+    bool withBaseRecord(const std::string& id, const std::function<bool(std::vector<std::string>&)>& function)

```

Comment:
The template argument would just resolve to the same thing though so all that'd do is save a bit of typing, right?

Reply:
Not only. `std::function` may not be optimized out https://godbolt.org/z/Qm84mz.

Reply:
Then it won't get optimised out unless I also replace the member function of the same name, since the argument is already an std::function at that point. Unless I'm missing something.

Reply:
You're right, I've missed that.